src/plugins/intel_cpu/src/nodes/executors/subgraph.cpp

```diff
@@ -61,7 +61,14 @@ SubgraphCodeGenerator::SubgraphCodeGenerator(const std::shared_ptr<SubgraphAttrs
     OPENVINO_ASSERT(config, "Runtime Config is empty!");
 
     jit_snippets_compile_args jcp;
-    jcp.data_offsets = config->io_data_offsets;
+    jcp.data_offsets.reserve(config->io_data_offsets.size());
+    const auto& brgemm_external_ptrs_idces = config->brgemm_external_ptrs_idces;
+    for (size_t i = 0; i < config->io_data_offsets.size(); ++i) {
+        // Note: external ptrs must be ignored by the kernel during compilation
+        if (!brgemm_external_ptrs_idces.count(i)) {
+            jcp.data_offsets.push_back(config->io_data_offsets[i]);
+        }
```

Comment:
In general I think that there might be situation when some offsets should be applied to `external ptrs`. For example, when this input has shape with rank greater than tile rank.

Could you please please check that all `fusion` passes fuses only when input `shape` has supported rank?
Also can we add the additional assert before this function call? For example, [here](https://github.com/openvinotoolkit/openvino/blob/6bc3b7ae87c52139e8ba455dc1b5cb190b86cf7b/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/fuse_brgemm_cpu_postops.cpp#L85).

Reply:
> In general I think that there might be situation when some offsets should be applied to external ptrs. For example, when this input has shape with rank greater than tile rank.

Do you mean a case when postop input rank is greater than tile rank and batch dims are not equal to 1? Currently, fusion of such postops is disabled ([code](https://github.com/v-Golubev/openvino/blob/vg/snippets/support_brgemm_postops/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/fuse_brgemm_cpu_postops.cpp#L284)), so external ptrs are always not incrementable